subsys/bluetooth/host/gatt.c

```diff
@@ -4757,7 +4757,14 @@ static int ccc_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 		} else if (!next) {
 			load.addr_with_id.id = BT_ID_DEFAULT;
 		} else {
-			load.addr_with_id.id = strtol(next, NULL, 10);
+			uint8_t id = strtol(next, NULL, 10);
+
+			if (!(id < CONFIG_BT_ID_MAX)) {
+				BT_ERR("Invalid local identity %d", id);
+				return -EINVAL;
+			}
+
+			load.addr_with_id.id = id;
```

Comment:
Prefer to use proper types when doing string operations as well as avoid negations in boolean expression if can be avoided. 

```suggestion
			unsigned long id = strtoul(next, NULL, 10);

			if (id >= CONFIG_BT_ID_MAX)) {
				BT_ERR("Invalid local identity %lu", id);
				return -EINVAL;
			}

			load.addr_with_id.id = (uint8_t)id;
```